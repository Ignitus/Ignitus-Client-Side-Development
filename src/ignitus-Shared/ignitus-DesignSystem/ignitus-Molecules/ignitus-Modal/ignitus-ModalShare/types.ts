export type ModalShareProps = {
  title: string;
  Share: string;
  Link: string;
  LinkedinColor: string;
  TwitterColor: string;
  RedditColor: string;
  FacebookColor: string;
  White: string;
  GreyOne: string;
};
