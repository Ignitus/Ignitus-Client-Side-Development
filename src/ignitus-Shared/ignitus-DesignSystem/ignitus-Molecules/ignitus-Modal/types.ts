export type Props = {
  secondary?: boolean;
};

export type ModalProps = {
  primaryText: string;
  secondaryText: string;
  title: string;
  buttonText: string;
};
