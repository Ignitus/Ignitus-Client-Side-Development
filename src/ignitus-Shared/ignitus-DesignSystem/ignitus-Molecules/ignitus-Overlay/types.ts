export type OverlayProps = {
  primaryText: string;
  secondaryText: string;
};
