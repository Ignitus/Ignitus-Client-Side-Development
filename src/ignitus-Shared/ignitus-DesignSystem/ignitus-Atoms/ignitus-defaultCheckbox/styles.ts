import styled from '@emotion/styled';
import { Paragraph } from '../typography';
import { LabelType } from './types';

export const CheckBox = styled.input`
  height: 1rem;
  width: 1rem;
  outline: none;
`;
export const Label = styled.label`
  margin-left: 0.4rem;
`;

export const StyledParagraph = styled(Paragraph)<LabelType>`
  font-size: ${props => props.size};
`;

export const Container = styled.div`
  display: inline-flex;
  align-items: center;
`;
