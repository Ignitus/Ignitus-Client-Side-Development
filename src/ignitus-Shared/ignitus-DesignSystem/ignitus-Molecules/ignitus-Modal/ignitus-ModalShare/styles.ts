import styled from '@emotion/styled';
import { Paragraph as P, Normal, MD, White, boxShadowColor } from '../../../..';
import { Icon } from '../../../../ignitus-Utilities/Components/icon';

import { flexibleColDiv, flexibleRowDiv } from '../../../shared';

export const ModalShare = styled.div`
  background-color: ${White};
  background-size: contain;
  border-radius: 1rem;
  display: inline-flex;
  flex-direction: column;
  box-shadow: 0 2px 4px 0 ${boxShadowColor};
  width: 35rem;
`;

export const TopSection = styled(flexibleRowDiv)`
  padding: 1.5rem;
  justify-content: space-between;
`;

export const BottomSection = styled(flexibleColDiv)`
  padding: 0.4rem 0.8rem 1.5rem 1.5rem;
  align-items: flex-start;
`;

export const IconsContainer = styled(flexibleRowDiv)`
  border-radius: 1rem;
  display: inline-flex;
`;

export const IconContainer = styled.button`
  border: none;
  cursor: pointer;
  overflow: hidden;
  margin: 0 0.2rem;
  height: 2.8rem;
  width: 2.8rem;
  border-radius: 50%;
  background-color: ${props => props.color};
  outline-color: transparent;
`;

export const Paragraph = styled(P)`
  font-weight: ${Normal};
  font-size: ${MD};
  margin: 2rem 0.6rem 0.6rem 0.6rem;
`;

export const StyledIcon = styled(Icon)`
  width: 1.4rem;
  fill: ${props => props.color};
  transition: 0.3s;
  flex: 1;
  margin-top: 0.2rem;
`;
