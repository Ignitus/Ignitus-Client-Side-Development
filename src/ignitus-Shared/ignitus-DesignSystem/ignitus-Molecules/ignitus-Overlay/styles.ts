import styled from '@emotion/styled';
import { flexibleColDiv } from '../../shared';
import { Icon } from '../../../ignitus-Utilities/Components/icon';
import { Heading1 } from '../../..';

export const Overlay = styled.div`
  position: relative;
  display: flex;
  align-items: center;
  justify-content: center;
  background-color: rgba(0, 0, 102, 0.89);
  backdrop-filter: blur(10px);
  width: 100%;
  height: 100%;
`;

export const Logo = styled(Icon)`
  margin-bottom: 4rem;
  height: 7.5rem;
  path {
    fill: white;
  }
`;

export const Heading = styled(Heading1)`
  margin-bottom: 1.5rem;
`;

export const MessageSection = flexibleColDiv;
