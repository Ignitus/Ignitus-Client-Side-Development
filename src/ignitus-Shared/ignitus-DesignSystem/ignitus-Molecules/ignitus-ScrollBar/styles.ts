import styled from '@emotion/styled';

export const Scroll = styled.div`
  margin: 30px;
  float: left;
  height: 300px;
  width: 65px;
  background: #f5f5f5;
  overflow-y: scroll;
  margin-bottom: 25px;

  ::-webkit-scrollbar {
    width: 12px;
    background: #f5f5f5;
  }

  ::-webkit-scrollbar-track {
    box-shadow: inset 0 0 6px rgba(0, 0, 0, 0.3);
    border-radius: 10px;
    background-color: #f5f5f5;
  }

  ::-webkit-scrollbar-thumb {
    border-radius: 10px;
    background-color: #f90;
  }
`;

export const Overflow = styled.div`
  min-height: 450px;
`;
