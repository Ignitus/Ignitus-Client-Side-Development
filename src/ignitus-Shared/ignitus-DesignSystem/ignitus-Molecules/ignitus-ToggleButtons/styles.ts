import styled from '@emotion/styled';

import { Icon } from '../../../ignitus-Utilities/Components/icon';
import * as B from '../../ignitus-Atoms/buttons';

export const ButtonLeft = styled(B.Button)`
  font-weight: 600;
  margin: 0;
  border: 0;
  float: left;
  border-radius: 20px 0 0 20px;
`;

export const ButtonRight = styled(B.Button)`
  font-weight: 600;
  margin: 0;
  border: 0;
  float: left;
  border-radius: 0 20px 20px 0;
`;

export const ToggleButtonContainer = styled.span`
  cursor: pointer;
`;

export const StyledIcon = styled(Icon)`
  width: 1.4rem;
  fill: ${props => props.color};
  transition: 0.3s;
  flex: 1;
  margin: 0.03rem;
`;
