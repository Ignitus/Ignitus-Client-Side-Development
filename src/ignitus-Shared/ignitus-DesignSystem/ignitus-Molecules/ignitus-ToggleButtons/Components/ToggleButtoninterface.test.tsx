import React from 'react';
import { mount, render } from 'enzyme';
import emotionSerializer, { matchers } from 'jest-emotion';

import { ToggleButtonInterface } from './ToggleButtoninterface';
import { ToggleButtonInterfaceProps } from '../types';
import { ButtonLeft, ButtonRight } from '../styles';

const props: ToggleButtonInterfaceProps = {
  activeCategory: 'primary',
  inActiveCategory: 'grey',
  children: {
    leftButtonContent: <> Left </>,
    rightButtonContent: <> Right </>,
  },
  handleClick: jest.fn(),
};

expect.addSnapshotSerializer(emotionSerializer);
expect.extend(matchers);

it('should render', () => {
  const tree = render(<ToggleButtonInterface {...props} />);
  expect(tree).toMatchSnapshot();
});

describe('<ToggleButtonInterface />', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = mount(<ToggleButtonInterface {...props} />);
  });

  it('should render buttons with correct categories', () => {
    expect(wrapper.find(ButtonLeft).prop('category')).toBe(
      props.activeCategory,
    );
    expect(wrapper.find(ButtonRight).prop('category')).toBe(
      props.inActiveCategory,
    );
  });

  it('should render buttons with correct content', () => {
    expect(wrapper.find(ButtonLeft).prop('children')).toBe(
      props.children.leftButtonContent,
    );
    expect(wrapper.find(ButtonRight).prop('children')).toBe(
      props.children.rightButtonContent,
    );
  });

  it('should simulate both side Clicks', () => {
    wrapper.find(ButtonLeft).simulate('click');
    expect(props.handleClick).toHaveBeenCalledWith('left');
    wrapper.find(ButtonRight).simulate('click');
    expect(props.handleClick).toHaveBeenCalledWith('right');
  });

  it('should toggle Button categories on click', () => {
    wrapper.find(ButtonLeft).simulate('click');
    expect(wrapper.find(ButtonLeft).prop('category')).toBe(
      props.activeCategory,
    );
    expect(wrapper.find(ButtonRight).prop('category')).toBe(
      props.inActiveCategory,
    );

    wrapper.find(ButtonRight).simulate('click');
    expect(wrapper.find(ButtonRight).prop('category')).toBe(
      props.activeCategory,
    );
    expect(wrapper.find(ButtonLeft).prop('category')).toBe(
      props.inActiveCategory,
    );
  });
});
