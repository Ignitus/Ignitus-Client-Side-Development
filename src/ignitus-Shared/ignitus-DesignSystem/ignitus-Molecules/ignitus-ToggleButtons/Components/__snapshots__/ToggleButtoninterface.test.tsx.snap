// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render 1`] = `
.emotion-2 {
  cursor: pointer;
}

.emotion-0 {
  border-radius: 5px;
  border: none;
  cursor: pointer;
  outline: none;
  padding: 16px 32px;
  text-align: center;
  line-height: 1rem;
  margin: 0.5rem;
  outline: none;
  font-size: 18px;
  font-weight: 400;
  -webkit-transition: all 0.3s ease;
  transition: all 0.3s ease;
  outline: none;
  color: #ffffff;
  background-color: #000066;
  padding: 8px 20px;
  font-size: 16px;
  border-radius: 3px;
  border: none;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-weight: 600;
  margin: 0;
  border: 0;
  float: left;
  border-radius: 20px 0 0 20px;
}

.emotion-0:hover,
.emotion-0:focus {
  box-shadow: 0 2px 4px 0 rgba(0,0,0,0.05);
}

.emotion-0 a {
  color: #ffffff;
}

.emotion-0:hover,
.emotion-0:focus {
  background-color: #F9BF64;
  color: #000066;
  outline: none;
}

.emotion-0:hover a,
.emotion-0:focus a {
  color: #000066;
}

.emotion-0:hover svg,
.emotion-0:focus svg {
  fill: #000066;
}

.emotion-1 {
  border-radius: 5px;
  border: none;
  cursor: pointer;
  outline: none;
  padding: 16px 32px;
  text-align: center;
  line-height: 1rem;
  margin: 0.5rem;
  outline: none;
  font-size: 18px;
  font-weight: 400;
  -webkit-transition: all 0.3s ease;
  transition: all 0.3s ease;
  outline: none;
  color: #000066;
  background-color: #E0E0E0;
  padding: 8px 20px;
  font-size: 16px;
  border-radius: 3px;
  border: none;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-weight: 600;
  margin: 0;
  border: 0;
  float: left;
  border-radius: 0 20px 20px 0;
}

.emotion-1:hover,
.emotion-1:focus {
  box-shadow: 0 2px 4px 0 rgba(0,0,0,0.05);
}

.emotion-1 a {
  color: #000066;
}

.emotion-1:hover,
.emotion-1:focus {
  background-color: #C4C4C4;
  color: #000066;
  outline: none;
}

.emotion-1:hover a,
.emotion-1:focus a {
  color: #000066;
}

.emotion-1:hover svg,
.emotion-1:focus svg {
  fill: #000066;
}

<span
  class="emotion-2"
>
  <button
    class="emotion-0"
  >
     Left 
  </button>
  <button
    class="emotion-1"
  >
     Right 
  </button>
</span>
`;
